ledger/participant-integration-api/src/main/resources/db/migration/postgres/V50__Append_only_schema.sql

```diff
@@ -0,0 +1,172 @@
+--
+-- TODO: reorder small fields to the end to avoid unnecessary padding.
+--
+DROP TABLE participant_events CASCADE;
+CREATE TABLE participant_events (
+    -- * kinds of events
+    event_kind smallint NOT NULL, -- Numbers allocated to leave some space for future additions.
+    -- 0:  divulgence event
+    -- 10: create event
+    -- 20: consuming exercise event
+    -- 25: non-consuming exercise event
+
+    -- * event identification
+    event_sequential_id bigserial NOT NULL, -- TODO temporarily readding bigserial for original write paths
+    -- NOTE: this must be assigned sequentially by the indexer such that
+    -- for all events ev1, ev2 it holds that '(ev1.offset < ev2.offset) <=> (ev1.event_sequential_id < ev2.event_sequential_id)
+
+    event_offset bytea,                                 -- NULL for divulgence events
+
+    -- * transaction metadata
+    transaction_id text,                                -- NULL for migrated divulgence events
+    ledger_effective_time timestamp without time zone,  -- NULL for migrated divulgence events
+    command_id text,
+    workflow_id text,
+    application_id text,
+    submitters text[],
+
+    -- * event metadata
+    node_index integer,  -- NULL for migrated divulgence events
+    event_id text,       -- NULL for migrated divulgence events
+    -- string representation of (transaction_id, node_index)
+
+    -- * shared event information
+    contract_id text NOT NULL,
+    template_id text,     -- TODO @simon@ with the new divulgance model supporting public pkv implementations: we need this to enable NULL-s. Do we need to make involved indexes partial?
+    flat_event_witnesses text[] DEFAULT '{}'::text[] NOT NULL,       -- stakeholders of create events and consuming exercise events
+    tree_event_witnesses text[] DEFAULT '{}'::text[] NOT NULL,       -- informees for create, exercise, and divulgance events
+
+    -- * divulgence and create events
+    create_argument bytea,
+
+    -- * create events only
+    create_signatories text[],
+    create_observers text[],
+    create_agreement_text text,
+    create_key_value bytea,
+    create_key_hash bytea,
+
+    -- * exercise events (consuming and non_consuming)
+    exercise_choice text,
+    exercise_argument bytea,
+    exercise_result bytea,
+    exercise_actors text[],
+    exercise_child_event_ids text[],
+
+    --compression flags
+    create_argument_compression SMALLINT,
+    create_key_value_compression SMALLINT,
+    exercise_argument_compression SMALLINT,
+    exercise_result_compression SMALLINT
+) PARTITION BY LIST (event_kind);
+
+
+-- Set storage parameters before creating partitions so they follow the pattern.
+-- these columns contain data that is generally incompressible, so don't try it
+ALTER TABLE participant_events ALTER COLUMN create_key_hash  SET STORAGE EXTERNAL;
+
+-- Partition the events according to the event type.
+-- TODO: manually partition the table in order to avoid using PostgreSQL-specific features
```

Comment:
Doing manual partitioning would mean that the `participant_events` table doesn't exist at all and the insertion happens directly to these tables, correct?